src/managers/config.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Manager for for handling configuration building + writing."""
+import logging
+from typing import TYPE_CHECKING
+
+from ops.model import ConfigData
+
+if TYPE_CHECKING:
+    pass
+
+from core.cluster import SUBSTRATES, ClusterState
+from core.workload import WorkloadBase
+
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_PROPERTIES = """
```

Comment:
I'm hesitating about this one given the level of complexity of the additional YAML parsing module vs. the current plain text implementation.

I clearly see the value of appending instead of erasing, however that may be possible to do still in a simple way.

Reply:
It's not simply about appending / erasing - though nice. But more importantly, it is to be able to read / load config options and have logic tied to current values, as well as atomically target config options for changes depending on variable input / certain logic in the charm.
This can be the subject of another PR, but in general, json or Yaml docs are hard to be accurately and easily dealt with as plain text, and should make use of the `yaml, json` libraries